shim/shim.c

```diff
@@ -39,9 +39,14 @@
 
 /* globals */
 
+// stdin is always added at index 3 in the poll array
+#define STDIN_INDEX 3
```

Comment:
Might be slightly safer to make this `(MAX_POLL_FDS-1)`?

I do wonder if we should change `add_pollfd()` so that it is told which index to assign the provided fd to make this code easier to maintain.